src/ScaleUnitManagement/WorkloadSetupOrchestrator/ScaleUnitWorkloadInstaller.cs

```diff
@@ -105,8 +105,9 @@ private async Task InstallWorkloadsOnScaleUnit()
                         await scaleUnitAosClient.WriteWorkloadInstances(workloadInstanceToInstallList);
                     }
 
-                    if (AxDeployment.IsApplicationVersionMoreRecentThan("10.8.581.0"))
+                    if (scaleUnit.EnvironmentType == EnvironmentType.LBD || AxDeployment.IsApplicationVersionMoreRecentThan("10.8.581.0"))
```

Comment:
Why is this check explicitly needed also for LBD environments? Is it not enough to have the app version check?

Reply:
Because AxDeployment.IsApplicationVersionMoreRecentThan("10.8.581.0") will try to read something from the file to detect the version. This implies that we are running the tool on the machine where the version is being detected.
`FileVersionInfo versionInfo = FileVersionInfo.GetVersionInfo($@"{scaleUnit.ServiceVolume}\AOSService\PackagesLocalDirectory\ApplicationSuite\bin\Microsoft.Dynamics.AX.DemandPlanning.Azure.dll");`

This will not work for LBD.

Should we just completely remove this check. It is not needed for versions older than 10.0.17.

Reply:
we should keep the check for the version, as we want the tool to be compatible with version 10.0.16 and above.

are we not going to run the tool on the machine where the version is being detected? if not how are we going to run step 1 (initialization of the hybrid topology)?

Reply:
I ran step 1 through Geneva action on Cloud environment and through powershell script on LBD environment.
So for CLoud-LBD we want to setup CLoud environment with this tool as well. So we cannot run the script from within either Cloud SF environment nor LBD environment.

Reply:
Got it. Thanks :)